crates/router/src/connector/bambora/transformers.rs

```diff
@@ -0,0 +1,217 @@
+};
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct BamboraPaymentsRequest {
+    amount: i64,
+    payment_method: String,
+    card: Card,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct BamboraPaymentsCaptureRequest {
+    amount: i64,
+    payment_method: String,
+    card: Card,
+}
+
+impl TryFrom<&types::PaymentsCaptureRouterData> for BamboraPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_item: &types::PaymentsCaptureRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            ..Default::default()
+        })
+    }
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for BamboraPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let payment_method_data = item.request.payment_method_data.clone();
+        let payment_method = match payment_method_data {
+            api::PaymentMethod::Card(ref _item) => String::from("card"),
+            _ => todo!(),
+        };
+        let card = match payment_method_data {
+            api::PaymentMethod::Card(ref item) => Card {
+                name: item.card_holder_name.peek().clone(),
+                number: item.card_number.peek().clone(),
+                expiry_month: item.card_exp_month.peek().clone(),
+                expiry_year: item.card_exp_year.peek().clone(),
+                cvd: item.card_cvc.peek().clone(),
+                complete: true,
+            },
+            _ => todo!(),
+        };
+        Ok(Self {
+            amount: item.request.amount,
+            payment_method,
+            card,
+        })
+    }
+}
+
+// Auth Struct
+pub struct BamboraAuthType {
+    pub(super) api_key: String,
+}
+
+impl TryFrom<&types::ConnectorAuthType> for BamboraAuthType {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        if let types::ConnectorAuthType::HeaderKey { api_key } = _auth_type {
+            Ok(Self {
+                api_key: api_key.to_string(),
+            })
+        } else {
+            Err(errors::ConnectorError::FailedToObtainAuthType.into())
+        }
+    }
+}
+// PaymentsResponse
+//TODO: Append the remaining status flags
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum BamboraPaymentStatus {
+    #[default]
+    ZERO,
+    ONE,
+}
+
+impl From<BamboraPaymentStatus> for enums::AttemptStatus {
+    fn from(item: BamboraPaymentStatus) -> Self {
+        match item {
+            BamboraPaymentStatus::ONE => Self::Charged,
+            BamboraPaymentStatus::ZERO => Self::Failure,
+        }
+    }
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct BamboraPaymentsResponse {
+    approved: BamboraPaymentStatus,
```

Comment:
Rename this field appropriately, it must signify that it is a status.

If Few fields are optional then keep them optional, This might fail if some fields are present only for specfic payment types. The deserialization might fail in that case

Also keep the fields that are only necessary, this will avoid unnecessary deserialization